### PR TITLE
Remove explicit dependency to X11 + add test on OpenGL

### PR DESCRIPTION
**Pull request by jcarpent**

### CODE REVIEW

**Review by jmirabel — APPROVED** [review]

**Review by nim65s — APPROVED** [review]
This works for me, thanks for the PR !

File: CMakeLists.txt [hunk 1]
```diff
@@ -191,11 +189,4 @@ INSTALL(FILES cmake-module/${PROJECT_NAME}-plugin.cmake
   DESTINATION lib/cmake/${PROJECT_NAME}
   PERMISSIONS OWNER_READ GROUP_READ WORLD_READ OWNER_WRITE)
 
-FOREACH(lib ${X11_INCLUDE_DIR})
-  PKG_CONFIG_APPEND_CFLAGS("-I${lib}")
-ENDFOREACH(lib ${X11_INCLUDE_DIR})
-PKG_CONFIG_APPEND_LIBS_RAW(${X11_LIBRARIES})
 PKG_CONFIG_APPEND_LIBS(${PROJECT_NAME})
-# }}}
```
[the remark this answers is not in the document]
Comment (jcarpent) [reply CMakeLists.txt:199]: > You shouldn't remove this fold marker but keep the opening one

Which one?

[the remark this answers is not in the document]
Comment (nim65s) [reply CMakeLists.txt:199]: This one is corresponding to the opening tag in 184. But I suppose we can keep those, and the vim indication, can't we ?

[the remark this answers is not in the document]
Comment (jmirabel) [reply CMakeLists.txt:199]: Yes, I think we can keep them.

[the remark this answers is not in the document]
Comment (jcarpent) [reply CMakeLists.txt:199]: Done!

**Pull request merged**